HIP/hip-336.md

```diff
@@ -291,8 +319,68 @@ The allowance adjustment transaction will fail under the following circumstances
 - The token referenced in an NFT allowance is fungible.
 - Duplicate crypto or fungible token allowances for a spender are specified.
 - Duplicate NFT serial number allowances for a spender are specified.
+- A negative NFT serial number is specified.
 - No allowances are specified. 
 
+### Add CryptoDeleteAllowance API
+
+A new Hedera API will be added under the Crypto Service called `CryptoDeleteAllowance`. This function is called by the token owner to delete an existing allowance.
+
+```protobuf
+// new message
+message CryptoWipeAllowance {
```

Comment:
As far as I understand, this API will remove all allowances to users (even if they are 20 or 100). I have doubts about this functionality since it directly enforces us not to allow huge amount of approvals. The more approvals we allow, the bigger the `for loop` will be for removing the allowances. I would argue that it is better to change the API so that those operations specify the `repeated` number of `spenders` for which you want to remove allowance. The same comment is true for `TokenWipeAllowance` as-well